google-cloudevent-types/src/main/java/com/google/events/cloud/audit/v1/AuditLog.java

```diff
@@ -0,0 +1,122 @@
+    private Map<String, Object> request;
+    private RequestMetadata requestMetadata;
+    private ResourceLocation resourceLocation;
+    private String resourceName;
+    private Map<String, Object> resourceOriginalState;
+    private Map<String, Object> response;
+    private Map<String, Object> serviceData;
+    private String serviceName;
+    private Status status;
+
+    /**
+     * Authentication information.
+     */
+    public AuthenticationInfo getAuthenticationInfo() { return authenticationInfo; }
+    public void setAuthenticationInfo(AuthenticationInfo value) { this.authenticationInfo = value; }
+
+    /**
+     * Authorization information. If there are multiple resources or permissions involved, then
+     * there is one AuthorizationInfo element for each {resource, permission} tuple.
+     */
+    public AuthorizationInfo[] getAuthorizationInfo() { return authorizationInfo; }
+    public void setAuthorizationInfo(AuthorizationInfo[] value) { this.authorizationInfo = value; }
+
+    /**
+     * Other service-specific data about the request, response, and other information associated
+     * with the current audited event.
+     */
+    public Map<String, Object> getMetadata() { return metadata; }
+    public void setMetadata(Map<String, Object> value) { this.metadata = value; }
+
+    /**
+     * The name of the service method or operation. For example
+     * "google.datastore.v1.Datastore.RunQuery"
+     */
+    public String getMethodName() { return methodName; }
+    public void setMethodName(String value) { this.methodName = value; }
+
+    /**
+     * The number of items returned from a List or Query API method, if applicable.
+     */
+    public Long getNumResponseItems() { return numResponseItems; }
+    public void setNumResponseItems(Long value) { this.numResponseItems = value; }
+
+    /**
+     * The operation request. This may not include all request parameters, such as those that
+     * are too large, privacy-sensitive, or duplicated elsewhere in the log record. It should
+     * never include user-generated data, such as file contents. When the JSON object
+     * represented here has a proto equivalent, the proto name will be indicated in the `@type`
+     * property.
+     */
+    public Map<String, Object> getRequest() { return request; }
+    public void setRequest(Map<String, Object> value) { this.request = value; }
+
+    /**
+     * Metadata about the operation.
+     */
+    public RequestMetadata getRequestMetadata() { return requestMetadata; }
+    public void setRequestMetadata(RequestMetadata value) { this.requestMetadata = value; }
+
+    /**
+     * The resource location information.
+     */
+    public ResourceLocation getResourceLocation() { return resourceLocation; }
+    public void setResourceLocation(ResourceLocation value) { this.resourceLocation = value; }
+
+    /**
+     * The resource or collection that is the target of the operation. For example
+     * "shelves/SHELF_ID/books"
```

Comment:
Long term, we usually don't like having strings for resourceNames, but have a method to build the name correctly.

Reply:
I don't think we can do this long-term unless we change the CAL API protos:

https://github.com/googleapis/google-cloudevents/blob/master/proto/google/events/cloud/audit/v1/data.proto#L155

The resource name is a string in the proto.

Reply:
Well, we could presumably modify qt if necessary. That said, we're not doing anything special here in the C# events library either - whereas we have strongly-typed resource name types in the C# *API* libraries.

I suspect this is fine - we'd want to be able to access the raw string anyway.

Reply:
Noted we can have special cases in our generators if we want to later.